lib/grape/middleware/versioner/accept_version_header.rb

```diff
@@ -17,45 +17,22 @@ module Versioner
       # X-Cascade header to alert Grape::Router to attempt the next matched
       # route.
       class AcceptVersionHeader < Base
-        def before
-          potential_version = (env[Grape::Http::Headers::HTTP_ACCEPT_VERSION] || '').strip
-
-          if strict? && potential_version.empty?
-            # If no Accept-Version header:
-            throw :error, status: 406, headers: error_headers, message: 'Accept-Version header must be set.'
-          end
+        include VersionerHelpers
 
-          return if potential_version.empty?
+        def before
+          potential_version = env[Grape::Http::Headers::HTTP_ACCEPT_VERSION]&.strip
+          throw_error_not_acceptable('Accept-Version header must be set.') if strict? && potential_version.blank?
 
-          # If the requested version is not supported:
-          throw :error, status: 406, headers: error_headers, message: 'The requested version is not supported.' unless versions.any? { |v| v.to_s == potential_version }
+          return if potential_version.blank?
 
+          throw_error_not_acceptable('The requested version is not supported.') unless potential_version_match?(potential_version)
```

Comment:
Probably `not_acceptable!` makes more sense (needs a bang and is similar to other throws).